optaplanner-examples/src/test/java/org/optaplanner/examples/nqueens/integration/constructionheuristic/ConstructionHeuristicTest.java

```diff
@@ -0,0 +1,201 @@
+package org.optaplanner.examples.nqueens.integration.constructionheuristic;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicPhaseConfig;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicType;
+import org.optaplanner.core.config.heuristic.selector.entity.EntitySorterManner;
+import org.optaplanner.core.config.heuristic.selector.value.ValueSorterManner;
+import org.optaplanner.core.config.phase.PhaseConfig;
+import org.optaplanner.core.config.score.definition.ScoreDefinitionType;
+import org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig;
+import org.optaplanner.core.config.solver.EnvironmentMode;
+import org.optaplanner.core.config.solver.SolverConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+import org.optaplanner.core.impl.solver.DefaultSolver;
+import org.optaplanner.examples.nqueens.domain.Column;
+import org.optaplanner.examples.nqueens.domain.NQueens;
+import org.optaplanner.examples.nqueens.domain.Queen;
+import org.optaplanner.examples.nqueens.domain.Row;
+import org.optaplanner.examples.nqueens.integration.util.QueenCoordinates;
+import org.optaplanner.examples.nqueens.integration.util.QueenCoordinatesStepListener;
+import org.optaplanner.examples.nqueens.solver.score.NQueensAdvancedIncrementalScoreCalculator;
+
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(Parameterized.class)
+public class ConstructionHeuristicTest {
+
+    private final ConstructionHeuristicType constructionHeuristicType;
+    private final EntitySorterManner entitySorterManner;
+    private final ValueSorterManner valueSorterManner;
+    private final List<QueenCoordinates> expectedCoordinates;
+
+    public ConstructionHeuristicTest(ConstructionHeuristicType constructionHeuristicType,
+                                     EntitySorterManner entitySorterManner, ValueSorterManner valueSorterManner,
+                                     List<QueenCoordinates> expectedCoordinates) {
+        this.constructionHeuristicType = constructionHeuristicType;
+        this.entitySorterManner = entitySorterManner;
+        this.valueSorterManner = valueSorterManner;
+        this.expectedCoordinates = expectedCoordinates;
+    }
+
+    @Test
+    public void testConstructionHeuristics() {
+        SolverConfig config = new SolverConfig();
```

Comment:
I'd prefer loading the nqueensSolverConfig.xml and then removing all solver phases, instead of having to declare score calculation etc here too. See ConstructionHeuristicTest#buildSolverFactory() for inspiration.